airbyte-oauth/src/main/java/io/airbyte/oauth/OAuthImplementationFactory.java

```diff
@@ -8,65 +8,65 @@
 import io.airbyte.config.StandardDestinationDefinition;
 import io.airbyte.config.StandardSourceDefinition;
 import io.airbyte.config.persistence.ConfigRepository;
-import io.airbyte.oauth.flows.AsanaOAuthFlow;
-import io.airbyte.oauth.flows.GithubOAuthFlow;
-import io.airbyte.oauth.flows.HubspotOAuthFlow;
-import io.airbyte.oauth.flows.IntercomOAuthFlow;
-import io.airbyte.oauth.flows.SalesforceOAuthFlow;
-import io.airbyte.oauth.flows.SlackOAuthFlow;
-import io.airbyte.oauth.flows.SnapchatMarketingOAuthFlow;
-import io.airbyte.oauth.flows.SurveymonkeyOAuthFlow;
-import io.airbyte.oauth.flows.TrelloOAuthFlow;
+import io.airbyte.oauth.flows.*;
 import io.airbyte.oauth.flows.facebook.FacebookMarketingOAuthFlow;
 import io.airbyte.oauth.flows.facebook.FacebookPagesOAuthFlow;
 import io.airbyte.oauth.flows.facebook.InstagramOAuthFlow;
 import io.airbyte.oauth.flows.google.GoogleAdsOAuthFlow;
 import io.airbyte.oauth.flows.google.GoogleAnalyticsOAuthFlow;
 import io.airbyte.oauth.flows.google.GoogleSearchConsoleOAuthFlow;
 import io.airbyte.oauth.flows.google.GoogleSheetsOAuthFlow;
+import java.lang.reflect.InvocationTargetException;
 import java.net.http.HttpClient;
 import java.util.Map;
-import java.util.UUID;
 
 public class OAuthImplementationFactory {
 
-  private final Map<String, OAuthFlowImplementation> OAUTH_FLOW_MAPPING;
+  private static final Map<String, Class<? extends OAuthFlowImplementation>> OAUTH_FLOW_MAPPING =
+      ImmutableMap.<String, Class<? extends OAuthFlowImplementation>>builder()
+          .put("airbyte/source-asana", AsanaOAuthFlow.class)
```

Comment:
what is the rationale behind this change? I think I prefer it less than the previous approach because it uses reflection, which means tracking how this code is being used is harder to do in the IDE. Also, reflection can fail at runtime, whereas the constructor usage is checked at compile time

Reply:
Yes, no very strong reasons. Personally I think this is more readable, but it is really personal preference. And your concerns are a good point, let me change it back to the way it was.